purchase_stock_picking_show_currency_rate/models/stock_picking.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    purchase_currency_id = fields.Many2one(
+        string="Purchase currency", related="purchase_id.currency_id", readonly=True,
+    )
+    company_currency_id = fields.Many2one(
+        string="Company Currency", related="company_id.currency_id", readonly=True,
+    )
+    show_currency_rate_amount = fields.Boolean(
+        compute="_compute_show_currency_rate_amount", readonly=True
+    )
+    currency_rate_amount = fields.Float(
+        string="Rate amount", compute="_compute_currency_rate_amount", digits=0,
+    )
+
+    @api.depends("purchase_currency_id", "company_currency_id")
+    def _compute_show_currency_rate_amount(self):
+        for item in self:
+            item.show_currency_rate_amount = False
+            if item.purchase_currency_id != item.company_id.currency_id:
+                item.show_currency_rate_amount = True
+
+    @api.depends("purchase_id", "show_currency_rate_amount", "company_id")
```

Comment:
Shouldn't these dependencies be like this?
```suggestion
    @api.depends("purchase_currency_id", "show_currency_rate_amount", "company_id")
```